### PR TITLE
[WIP] port of docs from deck.gl codebase

### PR DESCRIPTION
**Pull request by abmai**
It's working and semi-cleaned. Will continue to make diffs this week for more cleanup + dummy examples.

### CODE REVIEW

File: doc-pages/src/static/docs/whats-new.md [hunk 1]
```diff
@@ -0,0 +1,350 @@
+# deck.gl v4.0
+
+Release date: March 31, 2017
+
+## Highlights
+
+- **New Geospatial Layers** GeoJsonLayer, PathLayer, PolygonLayer, IconLayer,
```
Comment (ibgreen) [inline doc-pages/src/static/docs/whats-new.md]: Keep file, but cut down and add headers that make sense for react-map-gl

File: doc-pages/src/static/docs/upgrade-guide.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
+# Upgrade Guide
+
+## Upgrading from deck.gl v3 to v4
+
+### Changed Import: The `DeckGL` React component
+
+A small but breaking change that will affect all applications is that the
```
Comment (ibgreen) [inline doc-pages/src/static/docs/upgrade-guide.md]: Keep file, but cut down and add headers that make sense for react-map-gl

File: doc-pages/src/static/images/location-icon-mapping.json [hunk 3]
```diff
@@ -0,0 +1,142 @@
+{
```
Comment (ibgreen) [inline doc-pages/src/static/images/location-icon-mapping.json]: Cut

File: doc-pages/src/static/markdown/examples.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
+### Core Layers
```
Comment (ibgreen) [inline doc-pages/src/static/markdown/examples.md]: Keep file, but cut down and add headers that make sense for react-map-gl

File: doc-pages/src/static/docs/README.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
+<p align="right">
+  <a href="https://github.com/uber/deck.gl/blob/4.0-release/docs/README.md">
+    <img src="https://img.shields.io/badge/current-v4-brightgreen.svg?style=flat-square" />
+  </a>
+  <a href="https://github.com/uber/deck.gl/tree/3.0-release/docs">
+    <img src="https://img.shields.io/badge/legacy-v3-green.svg?style=flat-square" />
+  </a>
+</p>
+
+# Introduction
```
Comment (ibgreen) [inline doc-pages/src/static/docs/README.md]: Cut

**Comment by abmai** [comment]
Sounds good to me. I'm not entirely familiar with how github pages are built yet. Should I just go ahead with that and replace what's up on there right now?

**Comment by ibgreen** [comment]
> Sounds good to me. I'm not entirely familiar with how github pages are built yet. Should I just go ahead with that and replace what's up on there right now?

* I don't think we should touch github pages at this point, we likely have a number of iterations before we are ready.
* I was just suggesting the demo loads the markdown files for the docs from github (instead of from its local static files) to avoid having to copy them into the static folder of the demo server. Using links like these: `https://raw.githubusercontent.com/uber/react-map-gl/master/docs/README.md`

**Comment by abmai** [comment]
re: not touching the github pages

- Is there a good reason to keep the current github pages up? The docs are pretty outdated and don't really provide much value from my perspective. I'd rather put up a landing page and continue iterating in the background (e.g. have specific landing points to see the current working docs). This way we can use it to communicate to the world that we're planning a launch of V3 as well.

**Comment by ibgreen** [comment]
> Is there a good reason to keep the current github pages up? 

Hmm, not really, just didn't think that replacing them with something broken/unfinished made sense.
* The old docs have been up there for a long time, and the new docs are likely to be in pretty good shape within a week, so in that sense I don't see a hurry to update.
* That said, if we just make a clean landing page (perhaps saying we are working on a doc update) and don't show the WIP docs, then I think it would be fine to publish that.

**Comment by abmai** [comment]
landing this so smaller PRs can be made against this for better clarity

**Pull request merged**